nemo_curator/utils/distributed_utils.py

```diff
@@ -77,6 +82,81 @@ def get_filepath_without_extension(path: str) -> str:
     return filename
 
 
+def _worker_gpu_tuple() -> tuple[str, int]:
+    """
+    Runs on a Dask-CUDA worker.
+    Returns (hostname, gpu_index) where `gpu_index` is the index shown by `nvidia-smi`.
+    """
+
+    # Touch the GPU so a context is created (idempotent if one already exists)
+    cp.cuda.runtime.getDevice()
```

Comment:
This does not create a CUDA context, you can verify that by watching `nvidia-smi` while you launch a process that calls this. We canonically use `numba.cuda.current_context()` in Dask, I would suggest the same because that is already tested and known to work without any side-effects.